src/dotnet/commands/dotnet-compile-native/ILCompilerInvoker.cs

```diff
@@ -43,32 +37,29 @@ private void InitializeArgs(NativeCompileSettings config)
             var coreLibsPath = Path.Combine(config.IlcSdkPath, "sdk");
             foreach (var reference in Directory.EnumerateFiles(coreLibsPath, "*.dll"))
             {
-                argsList.Add($"-r");
-                argsList.Add($"{reference}");
+                argsList.Add($"-r:{reference}");
```

Comment:
Ah I see that you are dropping a response file. You may need to manually quote the piece after `-r:`
since this won't go through the argument escaper.

We do this in CSC. Recommend to write some test cases around spaces with paths here.